editor/src/components/canvas/multiplayer-cursors.tsx

```diff
@@ -184,3 +222,114 @@ const MultiplayerCursor = React.memo(
   },
 )
 MultiplayerCursor.displayName = 'MultiplayerCursor'
+
+const FollowingOverlay = React.memo(() => {
+  const colorTheme = useColorTheme()
+  const dispatch = useDispatch()
+
+  const self = useSelf()
+  const others = useOthers((list) => normalizeOthersList(self.id, list))
+
+  const following = useEditorState(
+    Substores.restOfEditor,
+    (store) => store.editor.multiplayer.following,
+    'FollowingOverlay following',
+  )
+  const canvasScale = useEditorState(
+    Substores.canvasOffset,
+    (store) => store.editor.canvas.scale,
+    'FollowingOverlay canvasScale',
+  )
+  const canvasOffset = useEditorState(
+    Substores.canvasOffset,
+    (store) => store.editor.canvas.roundedCanvasOffset,
+    'FollowingOverlay canvasOffset',
+  )
+  const mode = useEditorState(
+    Substores.restOfEditor,
+    (store) => store.editor.mode,
+    'FollowingOverlay mode',
+  )
+
+  const followed = React.useMemo(() => {
+    return others.find((other) => following != null && other.id === following.id)
+  }, [others, following])
+
+  React.useEffect(() => {
+    // when following another player, apply its canvas constraints
+    if (followed == null) {
+      if (following != null) {
+        // reset if the other player disconnects
+        dispatch([
+          updateMultiplayerState({ following: null }),
+          switchEditorMode(EditorModes.selectMode(null, false, 'none'), undefined, true),
+        ])
+      }
+      return
+    }
+
+    let actions: EditorAction[] = []
+    if (followed.presence.canvasScale != null && followed.presence.canvasScale !== canvasScale) {
+      actions.push(CanvasActions.zoom(followed.presence.canvasScale, null, true))
+    }
+    if (
+      followed.presence.canvasOffset != null &&
+      !pointsEqual(followed.presence.canvasOffset, canvasOffset)
+    ) {
+      actions.push(CanvasActions.positionCanvas(followed.presence.canvasOffset, true))
+    }
+    if (followed.presence.mode !== mode.type) {
+      actions.push(
+        updateMultiplayerState({
+          following: multiplayerStateFollowing(followed.id, followed.presence.mode),
+        }),
+      )
+      switch (followed.presence.mode) {
+        case 'live':
+          actions.push(switchEditorMode(EditorModes.liveMode(null), undefined, true))
+          break
+        default:
```

Comment:
I think it is better to list all modes explicitly and assertNever in the default, to make sure we don't forget about this when we introduce a new mode.